File: modified_scripts/python/matrix_multiplication/qwen2.5-coder_3b.py

def matrix_multiply(A, B):
    if len(A[0]) != len(B):
        raise ValueError("Number of columns in A must be equal to number of rows in B")
    
    result = [[0] * len(B[0]) for _ in range(len(A))]
    
    for i in range(len(A)):
        for j in range(len(B[0])):
            for k in range(len(B)):
                result[i][j] += A[i][k] * B[k][j]
    
    return result

File: modified_scripts/rust/matrix_multiplication/qwen2.5-coder_3b.rs
```rust
fn matrix_multiply(a: &[[i32; 4]; 4], b: &[[i32; 4]; 4]) -> [[i32; 4]; 4] {
    let mut result = [[0; 4]; 4];
    for i in 0..4 {
        for j in 0..4 {
            for k in 0..4 {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    result
}
```